pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1431,8 +1431,41 @@ type ObjectStoreUserSpec struct {
 	//The display name for the ceph users
 	// +optional
 	DisplayName string `json:"displayName,omitempty"`
+	//The additional capabilities for the ceph user
+	// +optional
+	// +nullable
+	Caps *CapSpec `json:"caps,omitempty"`
+	//The maximum bucket limit for the ceph user
+	// +optional
+	MaxBuckets int `json:"maxBuckets,omitempty"`
+	//The maximum size limit for the ceph user
+	// +optional
+	MaxSize string `json:"maxSize,omitempty"`
+	//The maximum object limit for the ceph user
+	// +optional
+	MaxObjects int `json:"maxObjects,omitempty"`
 }
 
+type CapSpec struct {
+	// +optional
+	User CapPerm `json:"user,omitempty"`
+	// +optional
+	Bucket CapPerm `json:"bucket,omitempty"`
+	// +optional
+	MetaData CapPerm `json:"metadata,omitempty"`
+	// +optional
+	Usage CapPerm `json:"usage,omitempty"`
+	// +optional
+	Zone CapPerm `json:"zone,omitempty"`
+}
+
+/*// CapType represents permission such as read, write for capability
+//+kubebuilder:validation:Enum=users;buckets;metadata;usage;zone
+type CapType string*/
+// CapPerm represents permission such as read, write for capability
+//+kubebuilder:validation:Enum=*;read;write
+type CapPerm string
```

Comment:
Not sure the kubebuilder validation will work indirectly, I'd suggest just use type `string` with the validation on the individual properties of the `CapSpec`. This may be why the CRDs validation is failing in the CI. Try generating the CRDs with `make crds` just to make sure.